File: apps/frontend/web/src/lib/api/user-management.ts
import { api } from './client'

export interface InviteUserRequest {
  email: string
  role: string
}

export interface UserInvitation {
  id: string
  tenant_id: string
  email: string
  role: string
  status: string
  expires_at: string
  created_at: string
  updated_at: string
}

export interface AcceptInvitationRequest {
  token: string
  password: string
  first_name: string
  last_name: string
}

export interface ResetPasswordRequest {
  email: string
}

export interface ConfirmResetPasswordRequest {
  token: string
  new_password: string
}

export interface ChangePasswordRequest {
  current_password: string
  new_password: string
}

export interface UpdateProfileRequest {
  first_name?: string
  last_name?: string
  avatar?: string
  phone?: string
  address?: string
  city?: string
  country?: string
  postal_code?: string
  date_of_birth?: string
  bio?: string
  language?: string
  timezone?: string
}

// User Invitation API
export const inviteUser = async (data: InviteUserRequest) => {
  const response = await api.post('/invitations', data)
  return response.data
}

export const getInvitations = async (): Promise<UserInvitation[]> => {
  const response = await api.get('/invitations')
  return response.data.invitations
}

export const acceptInvitation = async (data: AcceptInvitationRequest) => {
  const response = await api.post('/invitations/accept', data)
  return response.data
}

export const revokeInvitation = async (invitationId: string) => {
  const response = await api.delete(`/invitations/${invitationId}`)
  return response.data
}

export const resendInvitation = async (invitationId: string) => {
  const response = await api.post(`/invitations/${invitationId}/resend`)
  return response.data
}

// Password Management API
export const requestPasswordReset = async (data: ResetPasswordRequest) => {
  const response = await api.post('/password/reset/request', data)
  return response.data
}

export const confirmPasswordReset = async (data: ConfirmResetPasswordRequest) => {
  const response = await api.post('/password/reset/confirm', data)
  return response.data
}

export const changePassword = async (data: ChangePasswordRequest) => {
  const response = await api.post('/password/change', data)
  return response.data
}

// Email Verification API
export const sendVerificationEmail = async () => {
  const response = await api.post('/email/verify/send')
  return response.data
}

export const verifyEmail = async (token: string) => {
  const response = await api.post(`/email/verify?token=${token}`)
  return response.data
}

export const resendEmailVerification = async (data: { email: string }) => {
  const response = await api.post('/email/verify/resend', data)
  return response.data
}

// Profile Management API
export const getProfile = async () => {
  const response = await api.get('/profile')
  return response.data
}

export const updateProfile = async (data: UpdateProfileRequest) => {
  const response = await api.put('/profile', data)
  return response.data
}

File: apps/backend/auth-service/internal/routes/user_management.go
package routes

import (
	"zplus-saas/apps/backend/auth-service/internal/handlers"

	"github.com/gofiber/fiber/v2"
)

// SetupUserManagementRoutes sets up user management routes
func SetupUserManagementRoutes(app *fiber.App, handler *handlers.UserManagementHandler) {
	api := app.Group("/api")

	// User invitation routes
	invitations := api.Group("/invitations")
	invitations.Post("/", handler.InviteUser)                 // POST /api/invitations
	invitations.Get("/", handler.GetInvitations)              // GET /api/invitations
	invitations.Post("/accept", handler.AcceptInvitation)     // POST /api/invitations/accept?token=xxx
	invitations.Delete("/:id", handler.RevokeInvitation)      // DELETE /api/invitations/:id
	invitations.Post("/:id/resend", handler.ResendInvitation) // POST /api/invitations/:id/resend

	// Password management routes
	password := api.Group("/password")
	password.Post("/reset/request", handler.RequestPasswordReset) // POST /api/password/reset/request
	password.Post("/reset/confirm", handler.ResetPassword)        // POST /api/password/reset/confirm
	password.Post("/change", handler.ChangePassword)              // POST /api/password/change (authenticated)

	// Email verification routes
	email := api.Group("/email")
	email.Post("/verify/send", handler.SendVerificationEmail) // POST /api/email/verify/send (authenticated)
	email.Post("/verify", handler.VerifyEmail)                // POST /api/email/verify?token=xxx

	// User profile routes
	profile := api.Group("/profile")
	profile.Get("/", handler.GetProfile)    // GET /api/profile (authenticated)
	profile.Put("/", handler.UpdateProfile) // PUT /api/profile (authenticated)
}
